### PR TITLE
Fixing unit tests (discovered on Android).

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

I have observed that `Realm.Results.addListener` sometimes hangs on RN Android.

## ☑️ ToDos

* [x] 📝 Changelog entry
* [x] 🚦 Tests
* ~[ ] 📝 Public documentation PR created or is not necessary~
* ~[ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
* ~[ ] typescript definitions file is updated~
* ~[ ] jsdoc files updated~
* ~[ ] Chrome debug API is updated if API is available on React Native~

### CODE REVIEW

File: tests/js/results-tests.js [hunk 1]
```diff
@@ -449,7 +450,7 @@ module.exports = {
                 resolve();
             });
         }).then(() => {
-            return new Promise((r, _reject) => {
+            new Promise((r, _reject) => {
```
Comment (blagoev) [inline tests/js/results-tests.js]: why do we need this Promise at all

File: tests/js/list-tests.js [hunk 2]
```diff
@@ -41,7 +41,7 @@ module.exports = {
 
         TestCase.assertThrowsContaining(() => new Realm.List(), 'constructor');
 
-        TestCase.assertType(Realm.List, 'function');
+        //TestCase.assertType(Realm.List, 'function');
```
Comment (blagoev) [inline tests/js/list-tests.js]: if not needed remove it

File: tests/js/results-tests.js [hunk 3]
```diff
@@ -434,9 +434,10 @@ module.exports = {
         });
 
         let resolve, first = true;
-        return new Promise((r, _reject) => {
+        new Promise((r, _reject) => {
```
Comment (blagoev) [inline tests/js/results-tests.js]: we should return this promise

File: tests/js/results-tests.js [hunk 4]
```diff
@@ -434,9 +434,10 @@ module.exports = {
...
             resolve = r;
             realm.objects('TestObject').addListener((testObjects, changes) => {
+                console.log('FISK', testObjects.length, changes.insertions.length)
```
Comment (bmunkholm) [inline tests/js/results-tests.js]: this log statement should be removed.

**Comment by blagoev** [comment]
While working on something else I did reproduce this on iOS. 
Here is a change to the same test I did
```
testAddListener: function() {
        if (typeof navigator !== 'undefined' && /Chrome/.test(navigator.userAgent)) { // eslint-disable-line no-undef
            // FIXME: async callbacks do not work correctly in Chrome debugging mode
            return Promise.resolve();
        }

        const realm = new Realm({ schema: [schemas.TestObject] });
        realm.write(() => {
            realm.create('TestObject', { doubleCol: 1 });
            realm.create('TestObject', { doubleCol: 2 });
            realm.create('TestObject', { doubleCol: 3 });
        });

        let resolve = () => {};
        let first = true;

        realm.objects('TestObject').addListener((testObjects, changes) => {
            if (first) {
                TestCase.assertEqual(testObjects.length, 3);
                TestCase.assertEqual(changes.insertions.length, 0);
            }
            else {
                TestCase.assertEqual(testObjects.length, 4);
                TestCase.assertEqual(changes.insertions.length, 1);
            }
            first = false;
            resolve();
        });

        return new Promise((r, _reject) => {
                resolve = r;
                realm.write(() => {
                    realm.create('TestObject', { doubleCol: 1 });
                });
            });
    },
```

Could you review it so we can use it instead of the previous version which obviously fails on both iOS and Android

**Comment by kneth** [comment]
@blagoev I have updated the test.

**Comment by kneth** [comment]
I have been able to run tests locally.

**Pull request merged**